src/model.cpp

```diff
@@ -51,10 +52,22 @@ Model& Model::operator=(Model m)
 
 std::string Model::doSerialisation(FORMATS format) const
 {
+    // ImportMap
     typedef std::pair <std::string, std::string> ImportNamePair;
     typedef std::vector<ImportNamePair>::const_iterator VectorPairIterator;
     typedef std::map <ImportPtr, std::vector<ImportNamePair> > ImportMap;
     typedef ImportMap::const_iterator ImportMapIterator;
+    // VariableMap
+    typedef std::pair <VariablePtr, VariablePtr> VariablePair;
+    typedef std::vector<VariablePair>::const_iterator VariableMapIterator;
+    typedef std::vector<VariablePair> VariableMap;
+    VariableMap variableMap;
+    // ComponentMap
+    typedef std::pair <Component*, Component*> ComponentPair;
+    typedef std::vector<ComponentPair>::const_iterator ComponentMapIterator;
+    typedef std::vector<ComponentPair> ComponentMap;
```

Comment:
I'd probably change the code slightly for these to make it a bit cleaner, maybe :)
```c
typedef std::pair<Component*, Component*> ComponentPair;
typedef std::vector<ComponentPair> ComponentMap;
typedef ComponentMap::const_iterator ComponentMapIterator;
```